### PR TITLE
Implement data UTXO

### LINKED ISSUE #165: Add TTL and Data Size to UTXO
Update UTXO structure according to the latest changes of Yellow Paper.

### PR DESCRIPTION
**Pull request by rtsisyk**
- Split Output into MonetaryOutput and DataOutput
- Allow empty merkle tree and blocks without outputs
- Add "msg" command to CLI

Other changes:

- Fix error handling in CLI
- Print epoch changing notifications in CLI

Closes #165

### CODE REVIEW

**Comment by rtsisyk** [comment]
Re-based on the latest dev.

**Comment by rtsisyk** [comment]
Re-based on the top of #218

**Comment by rtsisyk** [comment]
I need this branch to start working on #168

File: blockchain/src/block.rs [hunk 1]
```diff
@@ -268,18 +268,26 @@ impl MonetaryBlock {
         // +\sum{P_i} + \sum{C_i} for i in txins
```
Comment (dbmcclain) [inline blockchain/src/block.rs]: I understand C_i as Pedersen commitment (an ECC point). But what is P_i?

Comment (rtsisyk) [reply blockchain/src/block.rs]: It's a typo, let me fix that.

Comment (rtsisyk) [reply blockchain/src/block.rs]: Fixed, thanks! 
Now  `// +\sum{C_i} for i in txins`.

**Pull request merged**